calibrate-touchscreen/src/main.rs

```diff
@@ -0,0 +1,429 @@
+    pointer_y: f64,
+    pointer_width: f64,
+    pointer_height: f64,
+}
+
+impl Dispatch<wl_registry::WlRegistry, ()> for State {
+    fn event(
+        state: &mut Self,
+        registry: &wl_registry::WlRegistry,
+        event: wl_registry::Event,
+        _: &(),
+        _: &Connection,
+        qh: &QueueHandle<Self>,
+    ) {
+        match event {
+            wl_registry::Event::Global { name, interface, version } => {
+                 match &interface[..] {
+                    "wl_output" => {
+                        let output = registry.bind::<wl_output::WlOutput, _, _>(name, 1, qh, ());
+                        state.outputs.insert(output, None);
+                    }
+                    "wl_compositor" => {
+                        let compositor =
+                            registry.bind::<wl_compositor::WlCompositor, _, _>(name, 1, qh, ());
+                        state.compositor = Some(compositor);
+                    }
+                    "wl_shm" => {
+                        let shm = registry.bind::<wl_shm::WlShm, _, _>(name, 1, qh, ());
+                        state.wl_shm = Some(shm);
+                    }
+                    "wl_seat" => {
+                        registry.bind::<wl_seat::WlSeat, _, _>(name, 1, qh, ());
+                    }
+                    "xdg_wm_base" => {
+                        let wm_base = registry.bind::<xdg_wm_base::XdgWmBase, _, _>(name, 1, qh, ());
+                        state.wm_base = Some(wm_base);
+                    }
+                    _ => {}
+                }
+            }
+            _ => {}
+        }
+    }
+}
+
+impl Dispatch<wl_output::WlOutput, ()> for State {
+    fn event(
+        state: &mut Self,
+        output: &wl_output::WlOutput,
+        event: wl_output::Event,
+        _: &(),
+        _: &Connection,
+        qh: &QueueHandle<Self>,
+    ) {
+        use wayland_client::protocol::wl_output::Mode;
+
+        match event {
+            wl_output::Event::Mode { flags: WEnum::Value(flags), width, height, refresh } => {
+                if flags.contains(Mode::Current) {
+                    state.outputs.entry(output.clone())
+                        .and_modify(|fullscreen| *fullscreen = Some(FullscreenSurface{
+                            width: width,
+                            height: height,
+                            wl_surface: None,
+                            buffer: None,
+                            xdg_surface: None,
+                    }));
+
+                    state.draw_fullscreen_surfaces(qh);
+                }
+            }
+
+            _ => {}
+        }
+    }
+}
+
+fn draw_target(tmp: &mut File, (buf_x, buf_y): (u32, u32), (target_x, target_y): (f64, f64)) {
+    let centre_x = (buf_x as f64 * target_x) as i64;
+    let centre_y = (buf_y as f64 * target_y) as i64;
+    let target_size = (max(buf_x, buf_y)/80) as i64;
+    use std::{io::Write};
+    let mut buf = std::io::BufWriter::new(tmp);
```

Comment:
`File` already [`impl`s` Write`](https://doc.rust-lang.org/std/fs/struct.File.html#impl-Write-for-%26File) so there's no need for the intermediate `BufWriter` here?